packages/cells/src/model.ts

```diff
@@ -810,6 +811,12 @@ export class CodeCellModel extends CellModel implements ICodeCellModel {
     if (this._outputs) {
       this._outputs.trusted = args.newValue as boolean;
     }
+    if (args.newValue as boolean) {
+      const codeCell = this.sharedModel as models.YCodeCell;
+      const metadata = codeCell.getMetadata();
+      metadata.trusted = true;
+      codeCell.setMetadata(metadata);
```

Comment:
Does it mean that if other users execute one cell, this cell on my notebook is trusted automatically?

Reply:
The feedback I get from my PR (https://github.com/jupyterlab/jupyterlab/pull/11494#issuecomment-979039559) is that in some contexts, trusting automatically, especially for cells containing JS code, can make users exposed to intended attacks

Reply:
I agree that user roles can be part of the picture here, for instance giving a user the right to execute cells implicitly means that this user is trusted.
But since we don't have roles yet, maybe your work in #11494 could be extended to all outputs, not only widgets?

Reply:
You're right, since users have the right to execute cells, they are trusted for their activites on the server-side. My PR prevents malicious activities on the frontend side, which mostly relates to the HTML/JS rendering. So I don't think we need to extend this to the `text/plain` output.

Reply:
But outputs can execute JavaScript too, if they are of type `text/html`. This is what the trust mechanism protects from. So I think your PR can be very relevant in this case too.

Reply:
For now, I only check the `application/*` type, but indeed I also need to take a look at the `text/html` type.